selectors/plugins.ts

```diff
@@ -17,6 +17,14 @@ export const getFilesDropdownPluginMenuItems = createSelector(
     },
 );
 
+export const getUserGuideDropdownPluginMenuItems = createSelector(
+    'getUserGuideDropdownPluginMenuItems',
+    (state: GlobalState) => state.plugins.components.UserGuideDropdown,
+    (components) => {
+        return (components || []) as unknown as UserGuideDropdownPluginComponent[];
```

Comment:
Adding the type as mentioned in the other comment I think should make this type juggling unnecessary.

There's also an initial state for the plugin components defined in another file, though it's sort of optional to initialize there since we can do the `components || []` thing here.